packages/sdk/src/across/lpFeeCalculator.ts

```diff
@@ -14,6 +14,7 @@ export default class LpFeeCalculator {
     this.blockFinder = new BlockFinder<Block>(provider.getBlock.bind(provider));
   }
   async getLpFeePct(tokenAddress: string, bridgePoolAddress: string, amount: BigNumberish, timestamp?: number) {
+    // TODO: Use `./rateModel` methods here instead of hardcoded constant.
```

Comment:
@daywiss what do you think is the best way to integrate `./rateModel.ts` into this method? One current issue right now is that `getAllRateModelEvents` expects a `Web3` contract object for `RateModelStore` to be passed in, while this method and the SDK in general expects an `ethers` contract. The `InsuredBridgeL1Client` and all code in `financial-templates-lib` expects to use `web3` providers, while the SDK uses `ethers` providers.

@mrice32 cc

Reply:
One really annoying implementation would be to implement two methods:
`getAllRateModelEventsWeb3` and `getAllRateModelEventsEthers`, or overload the function params and check for whether the provider/contract is web3/ethers. Thoughts?

Reply:
I ended up moving `getAllRateModelEvents` to the client, leaving the web3 event querying and reformatting up to the client before calling into the SDK to get the latest rate model for a block height.

Reply:
I also added an internal `createRateModelDictionary` helper method that standardizes how the rate model dictionary is constructed. This obfuscates the dictionary itself from the client of the `sdk`, leaving the only responsibility for the caller to slightly parse raw contract events into an array of `{blockNumber, rateModelString, tokenAddress}` objects.

Reply:
I think this is a good solution here.

However, long-term, we probably need a conversion function that will take a web3 provider and create an ethers provider from it. And a similar function in the opposite direction. The bots will never be able to comfortably share functionality with the FE through the SDK if we can't have some way of sharing code that interacts with the blockchain.